polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -22,3 +22,14 @@ use bitvec::vec::BitVec;
 
 /// Bit indices in the `HostConfiguration.node_features` that correspond to different node features.
 pub type NodeFeatures = BitVec<u8, bitvec::order::Lsb0>;
+
+/// Module containing feature-specific bit indices into the `NodeFeatures` bitvec.
+pub mod node_features {
+	/// Tells if tranch0 assignments could be sent in a single certificate.
+	/// Reserved for: https://github.com/paritytech/polkadot-sdk/issues/628
+	pub const ENABLE_ASSIGNMENTS_V2: u8 = 0;
+	/// First unassigned feature bit.
+	/// Every time a new feature flag is assigned it should take this value.
+	/// and this should be incremented.
+	pub const FIRST_UNASSIGNED: u8 = 1;
```

Comment:
why do we use this instead of the reading the length of the bitvec? is that for more compatibility in which we assume that if a new feature is set to 0, it should work the same on nodes that are not aware of the feature? shouldn't it be 0 in that case (assigned to 0 == unassigned?)? if not, why not use the bitvec length?

Reply:
Yeah you are right in the sense that this can be improved, but I don't think checking the bitlength vec is much better. What I suggest is to implement a check on the actual bits. If we assume that the happy case is nodes are up to date when we enable features, then we must check if all bits set in the runtime are known by the client.

Reply:
Also, in that case an `enum` would be better: variant index is the bit index.